src/Avalonia.OpenGL/Controls/OpenGlControlResources.cs

```diff
@@ -161,7 +161,9 @@ public async ValueTask DisposeAsync()
             }
 
             Surface.Dispose();
+           
             await _swapchain.DisposeAsync();
+            Context.Dispose();
```

Comment:
It can be dangerous if the context was created somewhere outside.
But looking at the source code, it is created only in TryCreate methods of this class. 
Can you make OpenGlControlBaseResources single constructor private just to be safe? It's not a public API, so nothing would be broken.

Reply:
yes